acceptance/cmd_server_test.go

```diff
@@ -0,0 +1,74 @@
+package acceptance_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/databricks/cli/internal/testcli"
+	"github.com/stretchr/testify/require"
+)
+
+func StartCmdServer(t *testing.T) *TestServer {
+	server := StartServer(t)
+	server.Handle("/", func(r *http.Request) (any, error) {
+		q := r.URL.Query()
+		args := strings.Split(q.Get("args"), " ")
+
+		var env map[string]string
+		require.NoError(t, json.Unmarshal([]byte(q.Get("env")), &env))
+
+		for key, val := range env {
+			defer Setenv(t, key, val)()
+		}
+
+		defer Chdir(t, q.Get("cwd"))()
+
+		c := testcli.NewRunner(t, context.Background(), args...)
+		c.Verbose = false
+		stdout, stderr, err := c.Run()
+		result := map[string]any{
+			"stdout": stdout.String(),
+			"stderr": stderr.String(),
+		}
+		exitcode := 0
+		if err != nil {
+			exitcode = 1
+		}
+		result["exitcode"] = exitcode
+		return result, nil
+	})
+	return server
+}
+
+// Chdir variant that is intended to be used with defer so that it can switch back before function ends.
+// This is unlike testutil.Chdir which switches back only when tests end.
+func Chdir(t *testing.T, cwd string) func() {
```

Comment:
Please add a dummy `t.Setenv` call so this function is not accidentally called from parallel test executions.

Reply:
nit: This function (and Setenv) can also be tricky to read since it relies on `defer` statement computing it's args inline when it is called.

A better pattern might be to use Chdir twice:
```
testutil.Chdir(t, dir)
defer testutil.Chdir(t, dir)
```

Reply:
> Can you elaborate? What t.Setenv call?

The testing library does not directly expose an interface to prevent parallel executions of tests. Using t.Setenv on a dummy env var is a way to prevent accidental parallel executions of  test functions.


From the docs:
> Because Setenv affects the whole process, it cannot be used in parallel tests or tests with parallel ancestors

Reply:
ah I see, did not know about this trick.

I think for this limited use (Chdir is used only once and right after other setenv calls) it's fine like this. If this is exported into general library, I'll make sure to have precautionary SetEnv call there.